packages/core/core/src/RequestTracker.js

```diff
@@ -1167,15 +1202,10 @@ export default class RequestTracker {
         opts,
       ),
     );
-    report({
-      type: 'cache',
-      phase: 'write',
-      total,
-      size: this.graph.nodes.length,
-    });
-    report({type: 'cache', phase: 'end', total, size: this.graph.nodes.length});
 
     await Promise.all(promises);
```

Comment:
How many concurrent promises will be typical here for a Jira cache? `Promise.all` with a large set, especially when writing files or doing other async IO stuff, can have sub-optimal performance. If the concurrency is large enough, you might have better results using `async/queue` with a concurrency limit set.

Reply:
> (We also have PromiseQueue in the utils for exactly this use case you haven't seen it in the codebase yet.)

Ah, yeah, it's even touched in this PR  😅 

I was thinking of Jira where I have used `async/queue` in the past for this..